tests/Doctrine/Tests/ORM/Functional/Ticket/GH10912Test.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+use function array_filter;
+use function array_values;
+use function strpos;
+
+class GH10912Test extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->setUpEntitySchema([
+            GH10912User::class,
+            GH10912Room::class,
+        ]);
+    }
+
+    public function testIssue(): void
+    {
+        $user = new GH10912User();
+        $room = new GH10912Room($user);
+        $user->rooms->add($room);
+
+        $this->_em->persist($room);
+        $this->_em->persist($user);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $userReloaded = $this->_em->find(GH10912User::class, $user->id);
+
+        $queryLog = $this->getQueryLog();
+        $queryLog->reset()->enable();
+
+        $this->_em->remove($userReloaded);
+        $this->_em->flush();
+
+        $queries = array_values(array_filter($queryLog->queries, static function ($entry) {
+            return strpos($entry['sql'], 'DELETE') === 0;
+        }));
+
+        self::assertCount(2, $queries);
+        self::assertSame('DELETE FROM GH10912Room WHERE id = ?', $queries[0]['sql']);
+        self::assertSame('DELETE FROM GH10912User WHERE id = ?', $queries[1]['sql']);
+    }
+}
+
+/** @ORM\Entity */
+class GH10912User
+{
+    /**
+     * @ORM\Id
+     * @ORM\Column(type="integer")
+     * @ORM\GeneratedValue
+     *
+     * @var int
+     */
+    public $id;
+
+    /**
+     * @ORM\OneToMany(targetEntity=GH10912Room::class, mappedBy="user", cascade={"remove"})
```

Comment:
If I remove `cascade: remove`, I'll get the error

> Doctrine\DBAL\Exception\ForeignKeyConstraintViolationException : An exception occurred while executing a query: SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails (`doctrine_tests`.`GH10912Room`, CONSTRAINT `FK_7FCB83EEA76ED395` FOREIGN KEY (`user_id`) REFERENCES `GH10912User` (`id`))

This is because we have a `GH10912Room` pointing to the user that is to be removed, the association is not nullable and we have no database-level `ON DELETE` setting.

TIL that `cascade` seemingly also works on the inverse side of an association (@derrabus )

Reply:
Adding `orphanRemoval`, as can be seen in

https://github.com/Dallas62/doctrine-orm-reproducer-deletion-issue/blob/6e14b758be3484273ddddd2366f465a3d2eead30/src/Entity/User.php#L33

does not make difference.